src/posix/platform/netif.cpp

```diff
@@ -920,36 +932,99 @@ static void processReceive(otMessage *aMessage, void *aContext)
     VerifyOrExit(write(sTunFd, packet, length) == length, perror("write"); error = OT_ERROR_FAILED);
 
 exit:
-    otMessageFree(aMessage);
-
     if (error != OT_ERROR_NONE)
     {
         otLogWarnPlat("[netif] Failed to receive, error:%s", otThreadErrorToString(error));
     }
 }
 
-static void processTransmit(otInstance *aInstance)
+static void processReceiveIp6(otMessage *aMessage, void *aContext)
 {
-    otMessage *message = nullptr;
-    ssize_t    rval;
-    char       packet[kMaxIp6Size];
-    otError    error  = OT_ERROR_NONE;
-    size_t     offset = 0;
+    OT_UNUSED_VARIABLE(aContext);
 
-    assert(gInstance == aInstance);
+    assert(gInstance == aContext);
+    processReceive(aMessage);
+    otMessageFree(aMessage);
+}
 
-    rval = read(sTunFd, packet, sizeof(packet));
-    VerifyOrExit(rval > 0, error = OT_ERROR_FAILED);
+#if OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE
+static void processReceiveIp4(otIp4Message *aMessage, void *aContext)
+{
+    OT_UNUSED_VARIABLE(aContext);
+
+    assert(gInstance == aContext);
+    processReceive(otCastIp4Message(aMessage));
+    otIp4MessageFree(aMessage);
+}
+#endif
 
+#if OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE
```

Comment:
Can remove this if?

Reply:
No, the compiler will complain about the unused sendIp4Message (since line 1047 is the only usage of this function.